src/Api/Utilities/ServiceCollectionExtensions.cs

```diff
@@ -69,4 +70,50 @@ public static void AddSwagger(this IServiceCollection services, GlobalSettings g
             config.IncludeXmlComments(coreFilePath);
         });
     }
+
+    public static void AddHealthChecks(this IServiceCollection services, GlobalSettings globalSettings)
+    {
+        services.AddHealthCheckServices(globalSettings, builder =>
+        {
+            var identityUri = new Uri(globalSettings.BaseServiceUri.Identity
+                                      + "/.well-known/openid-configuration");
+
+            builder.AddUrlGroup(identityUri, "identity");
+
+            if (!string.IsNullOrEmpty(globalSettings.SqlServer.ConnectionString))
+            {
+                builder.AddSqlServer(globalSettings.SqlServer.ConnectionString);
+            }
+
+            if (!string.IsNullOrEmpty(globalSettings.Redis.ConnectionString))
+            {
+                builder.AddRedis(globalSettings.Redis.ConnectionString);
+            }
+
+            if (!string.IsNullOrEmpty(globalSettings.Storage.ConnectionString))
+            {
+                builder.AddAzureQueueStorage(globalSettings.Storage.ConnectionString, name: "storage_queue")
+                    .AddAzureQueueStorage(globalSettings.Events.ConnectionString, name: "events_queue");
+            }
+
+            if (!string.IsNullOrEmpty(globalSettings.Notifications.ConnectionString))
+            {
+                builder.AddAzureQueueStorage(globalSettings.Notifications.ConnectionString,
+                    name: "notifications_queue");
+            }
+
+            if (!string.IsNullOrEmpty(globalSettings.ServiceBus.ConnectionString))
+            {
+                builder.AddAzureServiceBusTopic(_ => globalSettings.ServiceBus.ConnectionString,
+                    _ => globalSettings.ServiceBus.ApplicationCacheTopicName, name: "service_bus");
+            }
+
+            if (!string.IsNullOrEmpty(globalSettings.Mail.SendGridApiKey))
+            {
+                builder.AddSendGrid(globalSettings.Mail.SendGridApiKey);
+            }
+
+            builder.AddCheck<AmazonSesHealthCheck>("amazon_ses");
```

Comment:
This should also be behind an if check. Self host won't always have this config.